docs/prepare-your-dev-environment.md

```diff
@@ -69,21 +78,35 @@ Install the `libgpgme-dev` package.
     ```
 
 ## Getting Started
+
 1. Login to Azure:
+
     ```bash
     az login
     ```
 
 1. Clone the repository to your local machine:
+
     ```bash
     go get -u github.com/Azure/ARO-RP/...
     ```
+
     Alternatively you can also use:
+
     ```bash
     git clone https://github.com/Azure/ARO-RP.git $GOPATH/src/github.com/Azure/ARO-RP
     ```
 
 1. Go to project:
+
     ```bash
     cd ${GOPATH:-$HOME/go}/src/github.com/Azure/ARO-RP
     ```
+
+1. Install Go modules
+    1. Read comments in [update-go-module-dependencies.sh](../hack/update-go-module-dependencies.sh) for more information.
+
+    ```bash
+    go mod tidy
+    hack/update-go-module-dependencies.sh
```

Comment:
This step doesn't actually "install the go modules" for the application to build/compile. This hack script updates the modules when we upgrade our installer fork based on OCP version (it is a process going away _eventually_). Lets remove this part out of the prepare your env as it is covered in the "updating dependencies" document https://github.com/Azure/ARO-RP/blob/master/docs/updating-dependencies.md#update-the-whole-dependency-tree

Reply:
I think it should be removed all together. Updating the dependencies is not part of the normal development workflow. It is something that is usually only done once per quarter when we upgrade OCP versions. 

For example, this "one liner" is able to run `make generate` in a fresh go-toolset:1.17 container after `git clone`-ing the ARO-RP repo

```bash
docker run -it registry.access.redhat.com/ubi8/go-toolset:1.17 /bin/bash -c 'echo "" && uname -a && \
export GOPATH=$(pwd) && \
git clone https://github.com/Azure/ARO-RP.git $GOPATH/src/github.com/Azure/ARO-RP && \
cd $GOPATH/src/github.com/Azure/ARO-RP && \
make generate'
```

Reply:
Makes sense, I've removed it all together.